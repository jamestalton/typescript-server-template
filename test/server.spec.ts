import * as axios from 'axios'
import { setupBeforeAll, setupAfterAll } from './jest-setup'

let request: axios.AxiosInstance

beforeAll(() => {
    request = setupBeforeAll()
})

afterAll(setupAfterAll)

describe(`Service Routes`, function () {
    it(`GET / returns status 200 and hello world`, async function () {
        await expect(request.get(`/`)).resolves.toMatchObject({
            status: 200,
            headers: {
                'content-type': 'application/json',
                'content-length': '25',
            },
            data: {
                message: 'Hello World',
            },
        })
    })
})
