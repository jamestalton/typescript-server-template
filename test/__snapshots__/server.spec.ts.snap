// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Service Routes GET / returns status 200 and hello world 1`] = `
Object {
  "message": "Hello World",
}
`;
